blivet/devices/md.py

```diff
@@ -605,7 +569,33 @@ def _postCreate(self):
         for member in self.devices:
             member.format.mdUuid = self.uuid
 
-        self._removeStaleLVM()
+        def removeStaleLVM():
+            """ Remove any stale LVM metadata that pre-existed in a new array's on-disk footprint. """
+            log.debug("waiting 5s for activation of stale lvm on new md array %s", self.path)
+            time.sleep(5)
+            udev.settle()
+
+            try:
+                pv_info = lvm.pvinfo(device=self.path)[self.path]
+            except errors.LVMError as e:
+                return
+
+            vg_uuid = None
+            try:
+                vg_uuid = udev.device_get_vg_uuid(pv_info)
+            except KeyError:
+                return
+
+            if vg_uuid:
+                log.info("removing stale LVM metadata found on %s", self.name)
+                try:
+                    lvm.vgremove(None, vg_uuid=vg_uuid)
+                except errors.LVMError as e:
+                    log.error("Failed to remove stale volume group from newly-created md array %s: %s",
+                              self.path, str(e))
+                    raise
+
+        removeStaleLVM()
```

Comment:
This looks weird now that I see it. Maybe it does't have to be a separate function at all in the end? Anyway, looks good (or equally bad) to me either way.

Reply:
Yeah, it's gross either way. I just wanted to clearly separate this particular hack from the rest of the code.